ophyd/array.py

```diff
@@ -0,0 +1,61 @@
+from functools import reduce
+from ophyd import Device, DeviceStatus
+
+
+def ArrayDevice(devices, *args, **kwargs):
```

Comment:
Using a function here makes it so that it can't be subclassed.  This should be updated to fix that.